### PR TITLE
Move metrics to global variable

### PR DESCRIPTION
**Pull request by ferranbt**
# Description

This PR replaces the per-module `Metrics` struct with a global `metrics` module. The advantage is that now it is possible to emit at any place a `metric` to the outside while before, that variable had to be associated with a `Metric` struct and a module for which the lifecycle might be a bit more convoluted. 

For example, let's say that the `consensus` module has a sub-module `block-builder` to build the blocks and we want to expose metrics there. With this PR, you only need one line to export metrics inside `block-builder`.

The PR also replaces the `go-kit` framework with `go-metrics` which is the internal library it uses to export metrics.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### Manual tests

Please complete this section if you ran manual tests for this functionality, otherwise delete it

# Documentation update

Please link the documentation update PR in this section if it's present, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -47,7 +47,7 @@ test-e2e:
     # because it will get picked up and run during e2e tests
     # and the e2e tests should error out if any kind of race is found
 	go build -race -o artifacts/polygon-edge .
-	env EDGE_BINARY=${PWD}/artifacts/polygon-edge go test -v -timeout=30m ./e2e/...
+	env EDGE_BINARY=${PWD}/artifacts/polygon-edge go test -v -timeout=30m ./e2e/... -run TestIbft_TransactionFeeRecipient
```
Comment (Stefan-Ethernal) [inline Makefile]: This should be reverted I believe?

Comment (ferranbt) [reply Makefile]: Yep, thank you

File: network/server.go [hunk 2]
```diff
@@ -758,26 +754,22 @@ func (s *Server) SubscribeCh() (<-chan *peerEvent.PeerEvent, error) {
...
-		s.metrics.InboundConnectionsCount.Set(
-			float64(s.connectionCounts.GetInboundConnCount()),
-		)
+		metrics.SetGauge([]string{"inbound_connections_count"}, float32(s.connectionCounts.GetInboundConnCount()))
+
 	case network.DirOutbound:
-		s.metrics.OutboundConnectionsCount.Set(
-			float64(s.connectionCounts.GetOutboundConnCount()),
-		)
+		metrics.SetGauge([]string{"outbound_connections_count"}, float32(s.connectionCounts.GetInboundConnCount()))
```
Comment (vcastellm) [inline network/server.go]:
```suggestion
		metrics.SetGauge([]string{"outbound_connections_count"}, float32(s.connectionCounts.GetOutboundConnCount()))
```
copypasta error maybe?

**Review by vcastellm — APPROVED** [review]
LGTM aside from the suggestion.

File: network/server.go (hunk 2 above)
Comment (ferranbt) [reply network/server.go]: Indeed. Good catch.

File: txpool/txpool.go [hunk 3]
```diff
@@ -452,7 +448,7 @@ func (p *TxPool) Drop(tx *types.Transaction) {
 	clearAccountQueue(dropped)
 
 	// update metrics
-	p.metrics.PendingTxs.Add(float64(-1 * len(dropped)))
+	metrics.SetGauge([]string{"pending_transactions"}, float32(-1*len(dropped)))
```
Comment (Kourin1996) [inline txpool/txpool.go]: Original code added a value with the previous one. `SetGauge` can do same?

**Review by Kourin1996 — APPROVED** [review]
LGTM. I just have small questions.

File: consensus/ibft/ibft.go [hunk 4]
```diff
@@ -295,7 +294,7 @@ func (i *backendIBFT) startConsensus() {
 		}
 
 		// Update the No.of validator metric
-		i.metrics.Validators.Set(float64(i.currentValidators.Len()))
+		metrics.SetGauge([]string{"validators"}, float32(i.currentValidators.Len()))
```
Comment (JekaMas) [inline consensus/ibft/ibft.go:297]: Just curious, is it a thread-safe operation? Won't we get data races in tests?

Comment (ferranbt) [reply consensus/ibft/ibft.go:297]: No data races, it is thread-safe.

File: txpool/txpool.go (hunk 3 above)
Comment (ferranbt) [reply txpool/txpool.go]: Good catch, I will address it.

File: consensus/ibft/ibft.go (hunk 4 above)
Comment (JekaMas) [reply consensus/ibft/ibft.go:297]: I believe we don't have tests with race at the moment. Could we check?

Comment (ferranbt) [reply consensus/ibft/ibft.go:297]: I tested with the `txpool`, there are data races there but none related to the `metrics`. This library is used across many OSS projects (including Hashicorp tools) without explicit locks, so it is safely handled internally.

File: txpool/txpool.go (hunk 3 above)
Comment (ferranbt) [reply txpool/txpool.go]: Fixed 01888aacd724fbf29459a959f97100a2cace724c

**Review by Stefan-Ethernal — APPROVED** [review]

**Review by JekaMas — APPROVED** [review]

**Pull request merged**